test/integration/etcd_sharding_test.go

```diff
@@ -0,0 +1,133 @@
+// +build integration,etcd
+
+package integration
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/coreos/go-etcd/etcd"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	etcdtesting "k8s.io/kubernetes/pkg/storage/etcd/testing"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	"github.com/openshift/origin/pkg/cmd/server/api"
+	projectapi "github.com/openshift/origin/pkg/project/api"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+)
+
+func TestEtcdSharding(t *testing.T) {
+	// Start a separate etcd server
+	kubeEtcdServer := etcdtesting.NewEtcdTestClientServer(t)
+	defer kubeEtcdServer.Terminate(t)
+	kubeEtcdClient := etcd.NewClient(kubeEtcdServer.ClientURLs.StringSlice())
+
+	// Generate openshift configuration
+	openShiftMasterConfig, err := testserver.DefaultMasterOptions()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Configure openshift to write kubernetes data to a separate etcd
+	openShiftMasterConfig.KubernetesMasterConfig.EtcdClientInfo = &api.EtcdConnectionInfo{
+		URLs: kubeEtcdServer.ClientURLs.StringSlice(),
+	}
+
+	// Start the server
+	openShiftOptions := testserver.TestOptions{DeleteAllEtcdKeys: true, EnableControllers: true}
+	adminConfigFile, err := testserver.StartConfiguredMasterWithOptions(openShiftMasterConfig, openShiftOptions)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	adminConfig, err := testutil.GetClusterAdminClientConfig(adminConfigFile)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Create an openshift client
+	osClient, err := testutil.GetClusterAdminClient(adminConfigFile)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Create an etcd client to talk to the openshift etcd
+	osEtcdClient := etcd.NewClient(openShiftMasterConfig.EtcdClientInfo.URLs)
```

Comment:
use the helper function for getting an etcd client... this assumes the test etcd is unsecured